CHANGELOG.md

```diff
@@ -12,6 +12,7 @@
 - Prune `sample_reductions` when returning eval logs with `header_only=True`.
 - Improved error message for undecorated solvers.
 - For simple matching scorers, only include explanation if it differs from answer.
+- Add `llama-cpp-python` local model provider.
```

Comment:
I think we should call the provider `llama-cpp` rather than `llama-cpp-python` (as that's the name of the project, and python is implicit in the fact that this is a python lib, e.g. we don't call openai openai-python).

Reply:
I would usually agree with this, however in this case `llama-cpp-python` is the (confusing IMO) name of the project. It offers a web server on top of llama-cpp (that isn't just Python bindings for it).

Without this name, I'd worry that people would think llama.cpp is sufficient. But it actually requires the [llama-cpp-python OpenAI web server running](https://llama-cpp-python.readthedocs.io/en/latest/server/).

---

I realise the above is a bit confusing. Maybe the following comments help?
- [llama.cpp can run its own webserver](https://github.com/ggerganov/llama.cpp/blob/master/examples/server/README.md), however I believe this doesn't support logprobs. So people might assume `llama-cpp` would refer to this, rather than `llama-cpp-python`'s webserver (which behaves differently)
- `llama-cpp-python` is the service we're connecting to (which internally connects to llama.cpp). It is not the library we use to connect to `llama.cpp` from Inspect. This is maybe where the disanalogy from openai is - where we're connecting to the OpenAI service.

---

If after the above you'd still prefer to call it `llama-cpp` I'm happy to rename it to this though :)

Reply:
Okay, I see exactly what you are saying. llama-cpp-python is its own project w/ its own docs/requirements/capabilities. Given that I agree that we have the right name for this!